sa/storage-authority.go

```diff
@@ -673,6 +673,125 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 	return
 }
 
+const collectAuthIDs = `SELECT id FROM %s
+WHERE identifier = :ident AND
+status != :invalid AND
+status != :revoked AND
+expires > :now`
+
+// RevokeAuthorizationsByDomain invalidates all pending or finalized authorization
+// and any associated challenges for a single domain
+func (ssa *SQLStorageAuthority) RevokeAuthorizationsByDomain(ident core.AcmeIdentifier) (int64, int64, error) {
```

Comment:
This code needs some LIMITs and loops, throughout to give us a better sense of progress instead just repeated timeouts.

We've already got folks with lots and lots of pendingauths for single domains, for instance.

Also, just checking, but do all of these queries have a decent index for them? Nothing with filesort involved, right?